src/Widgets/GoogleMapsWidget/GoogleMapsWidgetComponent.js

```diff
@@ -7,129 +7,135 @@ import "./GoogleMapsWidget.scss";
 
 const maxWidth = 640;
 
-class GoogleMapsWidgetComponent extends React.Component {
-  constructor(props) {
-    super(props);
-
-    this.state = {
-      elementHeight: 0,
-      elementWidth: 0,
-      height: null,
-      width: null,
-    };
-
-    this.outerDivRef = React.createRef();
-
-    this.handleResize = this.handleResize.bind(this);
-  }
-
-  componentDidMount() {
-    this.handleResize();
-    window.addEventListener("resize", this.handleResize);
-  }
-
-  componentWillUnmount() {
-    window.removeEventListener("resize", this.handleResize);
+function GoogleMapsWidgetComponent(props) {
+  const address =
+    props.widget.get("address") || "Brandenburg Gate, Berlin, Germany";
+  const zoom = props.widget.get("zoom") || "15";
+  const apiKey = googleMapsApiKey();
+  const mapType = props.widget.get("mapType") || "static";
+
+  if (mapType === "static") {
+    return (
+      <StaticMap address={address} zoom={zoom} apiKey={apiKey}>
+        <Widgets widget={props.widget} mapType={mapType} />
+      </StaticMap>
+    );
   }
 
-  handleResize() {
-    const elementWidth = this.outerDivRef.current.offsetWidth;
-    const elementHeight = this.outerDivRef.current.offsetHeight;
-
-    if (
-      this.state.elementWidth !== elementWidth ||
-      this.state.elementHeight !== elementHeight
-    ) {
-      let width = elementWidth;
-      let height = elementHeight;
+  return (
+    <InteractiveMap address={address} zoom={zoom} apiKey={apiKey}>
```

Comment:
* [ ] FYI because we have two identical code paths that only differ in the outer component we could use a variable component, for example:

```jsx
const Map = (mapType === "static") ? StaticMap : InteractiveMap;
<Map ... />
```